polus-recycle-vector-plugin/src/main.py

```diff
@@ -0,0 +1,253 @@
+
+    args = parser.parse_args()
+    stitch_dir = args.stitch_dir
+    collection_dir = args.collection_dir
+    stitch_regex = args.stitch_regex
+    collection_regex = args.collection_regex
+    output_dir = args.output_dir
+    logger.info('stitch_dir = {}'.format(stitch_dir))
+    logger.info('collection_dir = {}'.format(collection_dir))
+    logger.info('stitch_regex = {}'.format(stitch_regex))
+    logger.info('collection_regex = {}'.format(collection_regex))
+    logger.info('output_dir = {}'.format(output_dir))
+    
+    logger.info("Parsing regular expressions...")
+    new_stitch_regex, stitch_variables = _parse_regex(stitch_regex)
+    new_collection_regex, collection_variables = _parse_regex(collection_regex)
+    logger.info("Parsed stitch regex: {}".format(new_stitch_regex))
+    logger.info("Parsed collection regex: {}".format(new_collection_regex))
+    
+    logger.info("Validating variables...")
+    _validate_vars(stitch_variables,collection_variables)
+    logger.info("Passed variable checks.")
+    
+    logger.info("Parsing stitching vectors...")
+    stitch_vector = _parse_stitch(stitch_dir,new_stitch_regex,stitch_variables)
+    logger.info("Found the following [z][t] combinations:")
+    zs = [key for key in stitch_vector.keys()]
+    zs.sort()
+    for z in zs:
+        ts = [key for key in stitch_vector[z].keys()]
+        ts.sort()
+        for t in ts:
+            logger.info("[z][t]: [{}][{}]".format(z,t))
+            
+    logger.info("Parsing files in collection...")
+    collection_index = _parse_files(collection_dir,new_collection_regex,collection_variables)
+    logger.info("Found the following [z][t][c] combinations:")
+    zs = [key for key in collection_index.keys()]
+    zs.sort()
+    for z in zs:
+        ts = [key for key in collection_index[z].keys()]
+        ts.sort()
+        for t in ts:
+            cs = [key for key in collection_index[z][t].keys()]
+            cs.sort()
+            for c in cs:
+                logger.info("[z][t][c]: [{}][{}][{}]".format(z,t,c))
+    
+    logger.info("Generating stitching vectors...")
+    out_index = 1
+    zs = [z for z in stitch_vector.keys()]
+    zs.sort()
+    for z in zs:
+        if z not in collection_index.keys():
+            continue
+        ts = [t for t in stitch_vector[z].keys()]
+        ts.sort()
+        for t in ts:
+            if t not in collection_index[z].keys():
+                continue
+            cs = [c for c in collection_index[z][t].keys()]
+            cs.sort()
+            for c in cs:
+                fname = "img-global-positions-{}.txt".format(out_index)
+                logger.info("Generating vector: {}".format(fname))
+                fout = os.path.join(output_dir,fname)
+                out_index += 1
+                _generate_stitch(fout,stitch_vector[z][t],collection_index[z][t][c])
+                
+    logger.info("Plugin completed all operations!")
+
+if __name__ == "__main__":
+    main()
```

Comment:
My only other questions goes back to a previous review as well. Do we also want to consider "replicate" as a relevant dimension in naming? In my opinion we do, but I'm open to arguement.